pandas/tests/test_base.py

```diff
@@ -1051,6 +1051,22 @@ def test_searchsorted(self):
             self.assertTrue(0 <= index <= len(o))
 
 
+class TestUInt64HashTable(tm.TestCase):
+
+    def test_lookup_overflow(self):
+        from pandas.hashtable import UInt64HashTable
+        xs = np.array([1, 2, 2**63], dtype=np.uint64)
+        m = UInt64HashTable()
+        m.map_locations(xs)
+        self.assert_numpy_array_equal(m.lookup(xs),
+                                      np.arange(len(xs), dtype=np.int64))
+
+    def test_get_unique(self):
+        s = pd.Series([1, 2, 2**63, 2**63], dtype=np.uint64)
```

Comment:
not on the hashtable itself. ideally you should add uint64 to lots of places that we test int64 (construction type things). IOW, to actually exercise this code. I know this is vague. happy to merge this and to do a followup with more tests (and potentially breaking things). later.